azurerm/internal/services/network/parse/virtual_hub_connection.go

```diff
@@ -6,20 +6,33 @@ import (
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 )
 
-type VirtualHubConnectionResourceID struct {
+type VirtualHubConnectionId struct {
 	ResourceGroup  string
 	VirtualHubName string
 	Name           string
 }
 
-func ParseVirtualHubConnectionID(input string) (*VirtualHubConnectionResourceID, error) {
+func NewVirtualHubConnectionID(id VirtualHubId, name string) VirtualHubConnectionId {
+	return VirtualHubConnectionId{
+		ResourceGroup:  id.ResourceGroup,
+		VirtualHubName: id.Name,
+		Name:           name,
+	}
+}
+
+func (id VirtualHubConnectionId) ID(subscription string) string {
```

Comment:
_minor:_ the other formatters use `subscriptionId` as the argument so we may as well make this consistent:

```suggestion
func (id VirtualHubConnectionId) ID(subscriptionId string) string {
```